airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java

```diff
@@ -72,32 +70,50 @@ public BootloaderApp(final Configs configs, final FeatureFlags featureFlags) {
    * @param configs
    * @param postLoadExecution
    */
-  public BootloaderApp(final Configs configs, final Runnable postLoadExecution, final FeatureFlags featureFlags) {
+  public BootloaderApp(final Configs configs,
+                       final Runnable postLoadExecution,
+                       final FeatureFlags featureFlags,
+                       final SecretMigrator secretMigrator) {
     this.configs = configs;
     this.postLoadExecution = postLoadExecution;
     this.featureFlags = featureFlags;
+    this.secretMigrator = secretMigrator;
   }
 
   public BootloaderApp() {
     configs = new EnvConfigs();
     featureFlags = new EnvVariableFeatureFlags();
-    postLoadExecution = () -> {
```

Comment:
Why do we have this postLoadExecution code block at all? It seems like it's just repeating some initialization from `load()` and then loading seed data, and I don't see much value in declaring this separately from the logic in `load()`. Am I missing something, or could we just move everything into the `load()` method?

Reply:
I am not sure what we do have the postLoadExecution extracted and not in the load method. I will try to gather context on that. In the meantime, I have move the migration after the loaddata in the postLoadExecution method. What do you mean by repeating intialization from `load`?

Reply:
> What do you mean by repeating intialization from load?
We create the `configDatabase`, `jsonSecretsProcessor`, and `configPersistence` objects in both this constructor and in the `load` method. This is just repeated code which seems unnecessary

Reply:
Ah I just saw that the `copySecrets` value for jsonSecretsProcessor differs between the two. But what I said goes for the other two

Reply:
Ok, I though you meant, that we were calling `configPersistence.loadData()` which I didn't noticed. I will look at the duplicated persistence layer initialization.